templates/al2023/provisioners/install-nvidia-driver.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+
+set -o pipefail
+set -o nounset
+set -o errexit
+
+if [ "$ACCELERATOR_VENDOR" != "nvidia" ]; then
+  exit 0
+fi 
+echo "Installing Nvidia ${NVIDIA_MAJOR_DRIVER_VERSION} drivers..."
+
+################################################################################
+### Add repository #############################################################
+################################################################################
+sudo dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/amzn2023/x86_64/cuda-amzn2023.repo
+sudo dnf config-manager --add-repo https://nvidia.github.io/libnvidia-container/stable/rpm/nvidia-container-toolkit.repo
+
+sudo sed -i 's/gpgcheck=0/gpgcheck=1/g' /etc/yum.repos.d/nvidia-container-toolkit.repo /etc/yum.repos.d/cuda-amzn2023.repo 
+
+################################################################################
+### Install drivers ############################################################
+################################################################################
+sudo mkdir -p /etc/dkms
+echo "MAKE[0]=\"'make' -j$(grep -c processor /proc/cpuinfo) module\"" | sudo tee /etc/dkms/nvidia.conf
+sudo dnf -y install kernel-modules-extra.x86_64
+
+function archive-open-kmods(){
+  sudo dnf --setopt=install_weak_deps=False -y module install nvidia-driver:${NVIDIA_MAJOR_DRIVER_VERSION}-open
```

Comment:
What is `--setopt=install_weak_deps=False` for?

Reply:
Gotcha. Not a bad idea but it ultimately depends on the maintainer of the RPM spec to mark things correctly. We don't have much/any control over the spec, so unless this saves a lot of space I'd err on just installing the weak dependencies 🤷

Reply:
I'll see how much we save, the AMI size dropped down after removing CUDA and the unneeded Neuron packages.